articles/ros_node_idl.md

```diff
@@ -97,59 +104,128 @@ This proposal has a number of potential upsides, but it also has some downsides
 ### This is only really useful if it gains significant adoption in upstream packages
 
 It's true that, if not all of the packages in one's system have adopted this, its gains are incomplete.
+
 However, it's still useful even if only a subset of the packages adopt it (e.g. one's own packages), which means even without significant upstream adoption it will still be useful to individuals or organizations.
+
 Upstream packages that haven't adopted this simply won't benefit from it.
+
 Also, its usefulness hopefully outweighs the work required to implement it upstream, and it's certainly something that can be contributed by community members given that the interface would be reviewed by the experts in the package.
 
 ### Declared and actual interface can get out of sync
 
 This is certainly a concern: an out-of-date interface is debatably less useful than having no interface at all.
+
 There are a number of possibilities that will help with this issue.
+
 One possibility is to more tightly couple the declared and actual interface by creating a library that consumes the declared interface and creates the corresponding ROS entities.
+
 Another is the fact that, as soon as the node is running, RCL itself (or another `ros2` command) can verify that the actual interface properly corresponds to the declared interface, and can act appropriately.
 
 ## Package interface
 
 How do upstream packages specify their interface requirements?
+
 Through a high-level description of all the actions, parameters, services and topics, provided or required, by each node within the package.
 
-The package interface is defined in a separate [XML][xml_wiki] file and exported from the `package.xml` using [REP 149's export mechanism][rep149_export], thereby avoiding pollution of the package manifest.
+The package interface is defined in a separate [XML][xml_wiki] file with suffix `.nodl.xml`. This XML file is exported to the [ament index][ament_index], either manually in the case of python projects or with a helper CMake macro.
+
 The interface may cover only a subset of nodes in a package, as long as the nodes that _are_ covered are done so completely.
 
-Here is an example IDL for a package containing two nodes:
+Here is an example NoDL for a package containing two nodes:
 
-{% include_relative ros2_node_idl/interface_declaration.xml %}
+{% include_relative ros2_nodl/interface_declaration.xml %}
 
-Once an IDL file is written, it is exported from the package manifest:
 
-{% include_relative ros2_node_idl/package.xml %}
+Once an NoDL file is written, it is exported from either `CMakeLists.txt` or `setup.py`.
```

Comment:
As @artivis suggests, I'd move the "for cmake based builds" bit into the section below and just give a hint here:

```suggestion
Once an NoDL file is written, it is exported from either `CMakeLists.txt` or `setup.py` (more details below).
```